drivers/rawexec/driver_unix_test.go

```diff
@@ -349,10 +342,21 @@ func TestRawExec_ExecTaskStreaming_User(t *testing.T) {
 	clienttestutil.RequireLinux(t)
 
 	d := newEnabledRawExecDriver(t)
+
+	// because we cannot set AllocID, see below
+	d.config.NoCgroups = true
+
 	harness := dtestutil.NewDriverHarness(t, d)
 	defer harness.Kill()
 
 	task := &drivers.TaskConfig{
+		// todo(shoenig) - Setting AllocID causes test to fail - with or without
+		//  cgroups, and with or without chroot. It has to do with MkAllocDir
+		//  creating the directory structure, but the actual root cause is still
+		//  TBD. The symptom is that any command you try to execute will result
+		//  in "permission denied" coming from os/exec. This test is imperfect,
+		//  the actual feature of running commands as another user works fine.
+		// AllocID: uuid.Generate()
```

Comment:
If we leave `AllocID` out then the `AllocDir.SharedDir` and `AllocDir.AllocDir` then both end up directly under the tempdirthen ends up directly under the tempdir (which is also `task.AllocDir`), instead of under the allocation ID one level down.

That is instead of `/tmp/nomad_driver_harness-/alloc/:alloc_id` and `/tmp/nomad_driver_harness-/alloc` we get `/tmp/nomad_driver_harness-/alloc` and `/tmp/nomad_driver_harness-` and maybe the latter has the wrong permissions now?